lib/datadog/ci/transport/http.rb

```diff
@@ -91,12 +103,32 @@ def adapter
           @adapter ||= Datadog::Core::Transport::HTTP::Adapters::Net.new(settings)
         end
 
-        # this is needed because Datadog::Tracing::Writer is not fully compatiple with Datadog::Core::Transport
-        # TODO: remove when CI implements its own worker
+        # adds compatibility with Datadog::Tracing transport and
+        # provides ungzipping capabilities
         class ResponseDecorator < ::SimpleDelegator
+          def payload
+            return @decompressed_payload if defined?(@decompressed_payload)
+
+            if gzipped?(__getobj__.payload)
+              Datadog.logger.debug("Decompressing gzipped response payload")
+              @decompressed_payload = Gzip.decompress(__getobj__.payload)
+            else
+              __getobj__.payload
+            end
+          end
+
           def trace_count
             0
           end
+
+          def gzipped?(payload)
+            return false if payload.nil? || payload.empty?
+
+            first_bytes = payload[0, 2]
+            return false if first_bytes.nil? || first_bytes.empty?
+
+            first_bytes.b == Datadog::CI::Ext::Transport::GZIP_MAGIC_NUMBER
```

Comment:
Not exactly, I need to convert the string to ASCII_BINARY before comparing bytes with `payload.b`, I am not sure if this operation takes constant time. Payload could be very large (for my biggest test repos it is about 10Mb, our customers have repos with many more tests, so it could be easily a 50Mb string)